app/src/main/java/com/arctouch/codechallenge/view/activities/DetailsActivity.java

```diff
@@ -0,0 +1,103 @@
+package com.arctouch.codechallenge.view.activities;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.arctouch.codechallenge.Controller;
+import com.arctouch.codechallenge.R;
+import com.arctouch.codechallenge.base.UILogHandler;
+import com.arctouch.codechallenge.model.Movie;
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder;
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.request.RequestOptions;
+
+public class DetailsActivity extends AppCompatActivity implements UILogHandler{
+
+    private Controller controller;
+    private final MovieImageUrlBuilder movieImageUrlBuilder = new MovieImageUrlBuilder();
+
+    private TextView tvLog;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
This method is quite big. Do you think there's an ideal size for method's/class's bodies? How does that relate to the "Single Responsibility" principle?

Reply:
My personal rule is that a method should be between 5 and 20 lines, but it slides a little when it comes to `OnCreate` methods in `Activities` and `Fragments`, because my intention is to link the `xml` elements to the variables and start some code initial values. The problem is that the IDE hides when the parameter is the implementation of a Class and sometimes I don't take these lines into account, leaving some `OnCreate` methods too big, what is a problem that I will fix about my coding behavior. It indeed relates to the "Single Responsibility" principle, as this method is linking variables with `xml` elements, setting values and implementing classes as method parameters and this should be divided in multiple methods.